src/commands/deploy.ts

```diff
@@ -1,155 +1,77 @@
+        url: flags.string({
+            description: 'where the app should be deployed to',
+        }),
+        username: flags.string({
+            char: 'u',
+            description: 'username to authenticate with',
+        }),
+        password: flags.string({
+            char: 'p',
+            description: 'password for the user',
+        }),
         token: flags.string({
-            char: 't', dependsOn: ['userid'],
+            char: 't',
             description: 'API token to use with UserID (instead of username & password)',
         }),
         userid: flags.string({
-            char: 'i', dependsOn: ['token'],
+            char: 'i',
             description: 'UserID to use with API token (instead of username & password)',
         }),
+        // flag with no value (-f, --force)
+        force: flags.boolean({ char: 'f', description: 'forcefully deploy the App, ignores lint & TypeScript errors' }),
+        update: flags.boolean({ description: 'updates the app, instead of creating' }),
+        code: flags.string({ char: 'c', dependsOn: ['username'], description: '2FA code of the user' }),
+        i2fa: flags.boolean({ description: 'interactively ask for 2FA code' }),
     };
 
     public async run() {
         const { flags } = this.parse(Deploy);
 
-        cli.action.start(`${ chalk.green('packaging') } your app`);
-
         const fd = new FolderDetails(this);
 
         try {
             await fd.readInfoFile();
         } catch (e) {
-            this.error(e && e.message ? e.message : e);
-            return;
-        }
-
-        const compiler = new AppCompiler(this, fd);
-        const report = compiler.logDiagnostics();
-
-        if (!report.isValid && !flags.force) {
-            this.error('TypeScript compiler error(s) occurred');
-            this.exit(1);
-            return;
+            this.error(e && e.message ? e.message : e, {exit: 2});
         }
-
-        const packager = new AppPackager(this, fd);
-        const zipName = await packager.zipItUp();
-
-        cli.action.stop('packaged!');
-
-        if (!flags.url) {
-            flags.url = await cli.prompt('What is the server\'s url (include https)?');
-        }
-
-        if (!flags.username && !flags.token) {
-            flags.username = await cli.prompt('What is the username?');
-        }
-
-        if (!flags.password && !flags.token) {
-            flags.password = await cli.prompt('And, what is the password?', { type: 'hide' });
-        }
-
         if (flags.i2fa) {
             flags.code = await cli.prompt('2FA code', { type: 'hide' });
         }
+        let serverInfo: INormalLoginInfo | IPersonalAccessTokenLoginInfo | {};
+        let zipName;
+        cli.log(chalk.bold.greenBright('   Starting App Deployment to Server\n'));
+        try {
+            cli.action.start(chalk.bold.greenBright('   Checking App Report'));
+            checkReport(this, fd, flags);
```

Comment:
Maybe this could happen right before the packaging message? It's not clear what the `Checking App Report` message means, and it is part of the "packaging" process anyhow